dask/dataframe/core.py

```diff
@@ -1272,15 +1332,14 @@ def ndim(self):
         """ Return dimensionality """
         return 2
 
-    @cache_readonly
-    def _dtypes(self):
-        """ for cache, cache_readonly hides docstring """
-        return self._get(self.dask, self._keys()[0]).dtypes
-
     @property
     def dtypes(self):
         """ Return data types """
-        return self._dtypes
+        if self._known_dtype:
```

Comment:
When do we not know the dtype?  What operations don't preserve dtype information, apply?